daemon/listener/db.js

```diff
@@ -0,0 +1,59 @@
+const { Pool } = require('pg')
+
+/*
+  Module to interface with Postgres database.
+ */
+
+const dbName = 'origin'
+const listingTable = 'listing'
+
+// TODO(franck): dynamically configure client.
+const pool = new Pool(
+  {
+    host: 'localhost',
+    database: dbName,
+    // user: 'franck',
+    // password: 'franck',
+  })
+
+
+async function getListing(listingId) {
+  return pool.query(`SELECT * FROM ${listingTable} WHERE id=$1`, [listingId]).then((res) => {
+    return res.rows
+  })
+}
+
+/*
+ * Returns all rows from the listing table.
+ * @throws Throws an error if the read operation failed.
+ * @returns A promise that resolves to a list of rows (can be empty).
+ *
+ * TODO(franck): add support for pagination.
+ */
+async function getListings() {
+  return pool.query(`SELECT * FROM ${listingTable}`, []).then((res) => {
```

Comment:
If you feel like it, you could use `await` in these instead of `then`. Might make things clearer and easier to read.

Reply:
Yes, good point. It is cleaner. Updated.